From 1e38207050cbcc069ffdea0297a6e6c7b3d7588d Mon Sep 17 00:00:00 2001
From: DarkLord1731 <darklord1731@gmail.com>
Date: Mon, 25 Jun 2018 09:22:17 +0530
Subject: [PATCH 17/21] libril: Force static radio capability

Force the ril to return REQUEST_NOT_SUPPORTED error on getRadioCapability request.
Android handles the error and makes static radio capability for us, based on our config_radio_access_family overlay.

* Fixes network mode switch
---
 libril/ril_service.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libril/ril_service.cpp b/libril/ril_service.cpp
index 6a60c0c..142d49a 100644
--- a/libril/ril_service.cpp
+++ b/libril/ril_service.cpp
@@ -2649,7 +2649,8 @@ Return<void> RadioImpl::getRadioCapability(int32_t serial) {
 #if VDBG
     RLOGD("getRadioCapability: serial %d", serial);
 #endif
-    dispatchVoid(serial, mSlotId, RIL_REQUEST_GET_RADIO_CAPABILITY);
+    RequestInfo *pRI = android::addRequestToList(serial, mSlotId, RIL_REQUEST_GET_RADIO_CAPABILITY);
+    sendErrorResponse(pRI, RIL_E_REQUEST_NOT_SUPPORTED);
     return Void();
 }
 
-- 
2.11.0

